File: engine/engines-list.py
from twelvelabs import TwelveLabs
import os

client = TwelveLabs(api_key=os.environ.get("API_KEY"))

engines = client.engine.list()
for i, engine in enumerate(engines, 1):
    print(f"Engine ID: {engine.id}")
    print(f"  Author: {engine.author}")
    print(f"  Allowed Engine Options: {engine.allowed_engine_options}")
    print(f"  Ready: {engine.ready}")
    print(f"  Finetune: {engine.finetune}")

File: engine/engines_list.js
import { TwelveLabs } from 'twelvelabs-js';

(async () => {
  const client = new TwelveLabs({ apiKey: process.env.API_KEY});

  const engines = await client.engine.list();
  engines.forEach((engine, index) => {
    console.log(`Engine ID ${engine.id}:`);
    console.log(`  Author: ${engine.author}`);
    console.log(`  Allowed engine options: ${engine.allowedEngineOptions}`);
    console.log(`  Ready: ${engine.ready}`);
    console.log(`  Finetune: ${engine.finetune}`);
  });
})();
